src/Components/Classes/index.jsx

```diff
@@ -1,127 +1,203 @@
   const [trainers, setTrainers] = useState([]);
+  const [readyToSendAndAdd, setReadyToSendAndAdd] = useState({
+    day: '',
+    hour: '',
+    activityId: '',
+    trainerId: '',
+    slots: ''
+  });
+  const [readyToSendAndEdit, setReadyToSendAndEdit] = useState({
+    day: '',
+    hour: '',
+    activityId: '',
+    trainerId: '',
+    slots: ''
+  });
+  const [currentId, setCurrentId] = useState('');
+  const [showEdit, setShowEdit] = useState(false);
+  const [showAdd, setShowAdd] = useState(false);
+
+  useEffect(() => {
+    getClasses();
+    getActivities();
+    getTrainers();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
   const getClasses = async () => {
     try {
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/classes/`);
-      const { message, data, error } = await response.json();
-      alert(message);
-
-      if (!error) {
-        setClasses(data);
-      }
+      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/classes/`);
+      const data = await res.json();
+      setClasses(data.data);
     } catch (error) {
       console.error(error);
     }
   };
 
   const getActivities = async () => {
     try {
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/activities/`);
-      const { message, data, error } = await response.json();
-      alert(message);
-
-      if (!error) {
-        setActivities(data);
-      }
+      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/activities/`);
+      const data = await res.json();
+      setActivities(data.data);
     } catch (error) {
       console.error(error);
     }
   };
 
   const getTrainers = async () => {
     try {
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers/`);
-      const { message, data, error } = await response.json();
-      alert(message);
-
-      if (!error) {
-        setTrainers(data);
-      }
+      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers/`);
+      const data = await res.json();
+      setTrainers(data.data);
     } catch (error) {
       console.error(error);
     }
   };
 
-  useEffect(() => {
-    getClasses();
-    getActivities();
-    getTrainers();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
-
-  const addItem = async (klass) => {
+  const addClass = async (readyToSendAndAdd) => {
     try {
-      const newClass = await fetch(`${process.env.REACT_APP_API_URL}/api/classes/`, {
+      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/classes/`, {
         method: 'POST',
-        body: JSON.stringify(klass),
+        body: JSON.stringify(readyToSendAndAdd),
         headers: {
           'Content-type': 'application/json'
         }
       });
-
-      const { message, data, error } = await newClass.json();
-      alert(message);
-
-      if (!error) {
-        setClasses([...classes, data]);
-      }
+      const data = await res.json();
```

Comment:
if here you try something like this:

        const newClass = {
        _id: data.data._id,
        activityId: {
            name: classes.find((item) => item._id === data.data.classId).activityId?.name
          },
        day: data.data.day,
        hour: data.data.hour,
        slots: data.data.slots
      };


      setClasses([...classes, newClass]);
    } catch (error) {
      console.error(error);
      
 i think it should work

Reply:
Even with that implementation the code behaves the same